src/main/java/codeu/controller/LoginServlet.java

```diff
@@ -84,6 +85,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     }
 
     request.getSession().setAttribute("user", username);
+    user.setLastLogin(Instant.now());
```

Comment:
This might need to be applied to other pages as well, imagine if the user stays logged in for 3 months, this will not be triggered. The user may be commenting on conversations still. 

Perhaps lastActive might be a better option that captures all actions. Where might be a better place to put this logic so that it applies to any part of the servlet? Better still, could we make all servlets in the app automatically have this functionality? Maybe we can have a `ChatHttpServlet` that is extended by all servlets that has this common logic.

Another hint on that: There might be cases where user do not exist, i.e logged out. We need to ignore these cases.

Reply:
Is this related to Activity?
This is what I've in mind:
Every time an activity is created you'll have to update `lastActive`.
Also, we could make some statistics there or other kind of data process, this would improve performance since we wouldn't need to recalculate anymore.

Reply:
This is more of knowing when was the last time the user was "active". Active here include not just posting but also viewing.